pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -3383,28 +3387,39 @@ func (c *Cloud) getInstancesByNodeNamesCached(nodeNames sets.String, states ...s
 
 func (c *Cloud) getInstancesByNodeNames(nodeNames []string, states ...string) ([]*ec2.Instance, error) {
 	names := aws.StringSlice(nodeNames)
+	ec2Instances := []*ec2.Instance{}
 
-	nodeNameFilter := &ec2.Filter{
-		Name:   aws.String("private-dns-name"),
-		Values: names,
-	}
+	for i := 0; i < len(names); i += filterNodeLimit {
```

Comment:
Might be worth splitting this into a `partition` function, just because it is tricky and it would be great to have a unit test on it.  I am surprised there isn't one already, but maybe I'm looking in the wrong place.